skbio/embedding/_protein.py

```diff
@@ -96,3 +123,80 @@ def __repr__(self):
 example_protein_embedding = ProteinEmbedding(
     np.random.randn(62, 1024),
     'IGKEEIQQRLAQFVDHWKELKQLAAARGQRLEESLEYQQFVANVEEEEAWINEKMTLVASED')
+
+
+class ProteinVector(SequenceVector):
+    r""" A vector representation of the protein sequence.
+
+    Parameters
+    ----------
+    sequence : str, Sequence, or 1D np.ndarray
+        Characters representing the protein sequence itself.
+    vector : np.ndarray
+        The vector representation of the protein sequence.
+
+    See Also
+    --------
+    Protein
+
+    Examples
+    --------
+    >>> from skbio.embedding import ProteinVector
+    >>> import numpy as np
+    >>> embedding = np.random.rand(10)
+    >>> sequence = "ACDEFGHIKL"
+    >>> ProteinVector(embedding, sequence)
+    ProteinVector
+    --------------------------
+    Stats:
+        length: 10
+        vector dimension: 10
+        has gaps: False
+        has degenerates: False
+        has definites: True
+        has stops: False
+    --------------------------
+    0 ACDEFGHIKL
+
+    """
+    default_write_format = "embed"
+
+    def __init__(
+        self, vector, sequence: str, **kwargs
+    ):
+
+        sequence = _validate_protein(sequence)
+        if len(vector.shape) == 1:
+            vector = vector.reshape(1, -1)
+        if len(vector.shape) == 2:
+            assert vector.shape[0] == 1, (
+                "Only 1 vector per sequence is allowed."
+            )
+        super(ProteinVector, self).__init__(
+            vector, sequence=sequence,  **kwargs
+        )
+
+    def __repr__(self):
+        r"""
+        Return a string representation of the ProteinVector object.
+
+        Returns
+        -------
+        str
+            A string representation of the ProteinEmbedding object.
+
+        See Also
+        --------
+        Protein
+        """
+        seq = Protein(str(self))
+
```

Comment:
There is a lot in common here with `SequenceVector.__repr__` and may be an opportunity to decompose